mlog/testing.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package mlog
+
+import (
+	"testing"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// testingWriter is an io.Writer that writes through t.Log
+type testingWriter struct {
+	tb testing.TB
+}
+
+func (tw *testingWriter) Write(b []byte) (int, error) {
+	tw.tb.Log(string(b))
```

Comment:
Let's trim off the newlines from the end of `b` here, as `t.Log` adds its own that clutters the output unnecessarily.

Reply:
Sure! :smile: